os/arch/arm/src/common/up_mpuinit.c

```diff
@@ -66,6 +106,20 @@
 
 void up_mpuinitialize(void)
 {
+	/* Initialize MPU register numbers' information */
+	mpu_regs_initialize(&g_mpuregs);
+
+	/* Maximum MPU region number sanity check */
+#ifdef CONFIG_ARMV7M_MPU
+	if (g_mpuregs.maxmpu_regions > CONFIG_ARMV7M_MPU_NREGIONS) {
+		DEBUGASSERT(g_mpuregs.maxmpu_regions < CONFIG_ARMV7M_MPU_NREGIONS);
+	}
+#elif defined(CONFIG_ARMV8M_MPU)
+	if (g_mpuregs.maxmpu_regions > CONFIG_ARMV8M_MPU_NREGIONS) {
+		DEBUGASSERT(g_mpuregs.maxmpu_regions < CONFIG_ARMV8M_MPU_NREGIONS);
+	}
+#endif
```

Comment:
Let's reduce duplicated code.
```c
#ifdef CONFIG_V7
#define MPU_MAX_NREGION    CONFIG_...V7M...
#elif defined (CONFIG_V8)
#define 
#endif 

void up_mpuinitialize(void)
{
    ....

    if (g_mpu_... >= MPU_MAX_NREGION) {
        debugassert
    }
}
```
1. debugassert is enough? not assert? What happen when we try to set invalid mpu region?
2. In the conditional, don't we need to '='? I think it needs.